packages/emotion-theming/types/index.d.ts

```diff
@@ -14,17 +18,43 @@ export function ThemeProvider<Theme>(
   props: ThemeProviderProps<Theme>
 ): React.ReactElement
 
-/**
- * @todo Add more constraint to C so that
- * this function only accepts components with theme props.
- */
-export function withTheme<C extends React.ComponentType<any>>(
-  component: C
-): React.SFC<AddOptionalTo<PropsOf<C>, 'theme'>>
+export function withTheme<P extends object>(
```

Comment:
Right now they are not synced - take a look at `component` type and what both say about `theme`

Reply:
I can't figure out how `EmotionTheming` could be used. The only use would be if you could export a module as an interface. For example

```
declare module "emotion-theming" extends EmotionTheming<MyTheme> { }
```
There is a discussion about this feature at https://github.com/microsoft/TypeScript/issues/420

Going to delete the type for now, it isn't referenced in documentation and a google search turns nothing up.

Reply:
Seems like it could be imported and used regularly, but I agree that it doesn't make much sense to export this like that. You can always use `typeof` to get those types

Reply:
Sorted out all the typings around this